crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -223,8 +223,10 @@ impl CoinReadApiServer for CoinReadApi {
             })?;
         Ok(Balance {
             coin_type: coin_type.to_string(),
-            coin_object_count: balance.num_coins,
-            total_balance: balance.balance,
+            coin_object_count: usize::try_from(balance.num_coins)
+                .map_err(|e| Error::SuiError(SuiError::ExecutionError(e.to_string())))?,
+            total_balance: u128::try_from(balance.balance)
+                .map_err(|e| Error::SuiError(SuiError::ExecutionError(e.to_string())))?,
```

Comment:
plz do not use execution error, it signals something very different (usually a VM error)

Reply:
done